src/ProbabilisticProgramming/Monad.dfy

```diff
@@ -7,6 +7,54 @@ module Monad {
   import Rand
   import Measures
 
+  export Spec
+    provides
+      Hurd,
+      Return,
+      Bind,
+      Coin,
+      Eval,
+      Rand,
+      Map,
+      Result,
+      Result.RestIn,
+      Result.In,
+      Result.Satisfies,
+      Result.Bind,
+      Result.Map,
+      Result.Equals,
+      Result.IsConverging,
+      Result.IsDiverging,
+      Result.ValueOf,
+      Result.RestOf,
+      DivergingResult,
+      ConvergingResult,
+      ResultsWithValueIn,
+      ResultsWithRestIn,
+      ResultEventSpace,
+      LiftInEventSpaceToResultEventSpace,
+      LiftRestInEventSpaceToResultEventSpace,
+      boolResultEventSpace,
+      natResultEventSpace,
+      Measures
+    reveals
+      IsAlwaysConverging
+
+  export extends Spec
+
+  export Loops extends Spec reveals Hurd, Result, Result.Bind, Result.Satisfies, Return, Bind
+
+  export UniformPowerOfTwo extends Spec reveals Hurd, Result, Bind, Result.Bind, Eval
+
+  export Uniform extends Spec reveals Hurd, Result
+
+  export Bernoulli extends Spec reveals Hurd, Result
+
+  export BernoulliExpNeg extends Spec reveals Hurd, Result
+
+  export Independence extends Spec reveals Hurd
```

Comment:
Why don't we have one export set `MonadInternals`? Not sure what the fine-grained approach buys us.